src/tsickle.ts

```diff
@@ -49,18 +49,32 @@ export interface AnnotatorHost {
 }
 
 /**
- * The header to be used in generated externs.  This is not included in the
+ * The header to be used in generated externs. This is not included in the
  * output of annotate() because annotate() works one file at a time, and
  * typically you create one externs file from the entire compilation unit.
+ *
+ * Note: we don't have suppress {duplicate} here but rather on the variable statement
+ * on purpose. See SUPPRESS_DUPLICATE_TAG.
  */
 export const EXTERNS_HEADER = `/**
  * @externs
- * @suppress {duplicate,checkTypes}
+ * @suppress {checkTypes}
  */
 // NOTE: generated by tsickle, do not edit.
 `;
 
 /**
+ * Always suppress duplicate definitions at the variable declaration,
+ * and not at the file level. This allows to prevent errors from duplicates
+ * no matter in which order closure evaluates the externs in the case
+ * that the colliding extern has no @suppress {duplicate} annotation.
```

Comment:
Can you clarify this comment a bit about what we discovered?  That is, something about "if the @suppress is at the file level, then it doesn't affect other instances of the variable in other files, while apparently applying it to the variable does".

Also, I wonder if we should just put the checktypes suppression in here too, so that they're all in one place?  What do you think?

Reply:
Updated the docs.

I would keep the `checktypes` suppression where it is for now, if that is ok with you. Emits less code.